sdk/core/Azure.Core.TestFramework/src/RecordedTestAttribute.cs

```diff
@@ -43,44 +43,73 @@ public override TestResult Execute(TestExecutionContext context)
+
+                        // If the recording succeeded, set an error result.
+                        if (context.CurrentResult.ResultState.Status == TestStatus.Passed)
+                        {
+                            context.CurrentResult.SetResult(ResultState.Error,
+                                "Test failed playback, but was successfully re-recorded (it should pass if re-run)." + Environment.NewLine +
+                                Environment.NewLine +
+                                originalResult.Message);
+                        }
+                        else
+                        {
+                            context.CurrentResult.SetResult(context.CurrentResult.ResultState,
+                                originalResult.Message, context.CurrentResult.StackTrace + Environment.NewLine + Environment.NewLine +
+                                                        "The [RecordedTest] attribute attempted to re-record, but failed: " +
+                                                        Environment.NewLine +
+                                                        context.CurrentResult.Message + Environment.NewLine);
+                        }
+
+                        // revert RecordTestMode to Playback
+                        SetRecordMode(context.TestObject as RecordedTestBase, RecordedTestMode.Playback);
                     }
-                    else
+                    else if (resultMessage.Contains(typeof(TestTimeoutException).FullName))
                     {
-                        context.CurrentResult.SetResult(context.CurrentResult.ResultState,
-                            originalResult.Message, context.CurrentResult.StackTrace + Environment.NewLine + Environment.NewLine +
-                            "The [RecordedTest] attribute attempted to re-record, but failed: " + Environment.NewLine +
-                            context.CurrentResult.Message + Environment.NewLine);
-                    }
+                        // retry once
+                        context.CurrentResult = context.CurrentTest.MakeTestResult();
+                        context.CurrentResult = innerCommand.Execute(context);
 
-                    // revert RecordTestMode to Playback
-                    SetRecordMode(context.TestObject as RecordedTestBase, RecordedTestMode.Playback);
+                        if (context.CurrentResult.ResultState.Status != TestStatus.Passed)
+                        {
+                            context.CurrentResult.SetResult(
+                                ResultState.Error,
+                                "The test timed out twice:" + Environment.NewLine +
+                                $"First attempt: {originalResult.Message}" + Environment.NewLine +
+                                $"Second attempt: {context.CurrentResult.Message}");
+                        }
+                        else
+                        {
+                            context.CurrentResult.SetResult(
```

Comment:
Should we include the original state here as well just to indicate how far out of the timeout we were?